mmv1/products/artifactregistry/Repository.yaml

```diff
@@ -116,3 +116,134 @@
         - :RELEASE
         - :SNAPSHOT
         default_value: :VERSION_POLICY_UNSPECIFIED
+  - !ruby/object:Api::Type::Enum
+    name: 'mode'
+    min_version: beta
+    description: |-
+      The mode configures the repository to serve artifacts from different sources.
+    input: true
+    values:
+      - :STANDARD_REPOSITORY
+      - :VIRTUAL_REPOSITORY
+      - :REMOTE_REPOSITORY
+    default_value: :STANDARD_REPOSITORY
+  - !ruby/object:Api::Type::NestedObject
+    name: 'virtualRepositoryConfig'
+    min_version: beta
+    conflicts:
+      - remote_repository_config
+    description: |-
+      Configuration specific for a Virtual Repository.
+    properties:
+      - !ruby/object:Api::Type::Array
+        name: 'upstreamPolicies'
+        description: |-
+          Policies that configure the upstream artifacts distributed by the Virtual
+          Repository. Upstream policies cannot be set on a standard repository.
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+            - !ruby/object:Api::Type::String
+              name: 'id'
+              description: |-
+                The user-provided ID of the upstream policy.
+            - !ruby/object:Api::Type::String
+              name: 'repository'
+              description: |+
+                A reference to the repository resource, for example:
+                "projects/p1/locations/us-central1/repository/repo1".
+            - !ruby/object:Api::Type::Integer
+              name: 'priority'
+              description: |-
+                Entries with a greater priority value take precedence in the pull order.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'remoteRepositoryConfig'
+    min_version: beta
+    conflicts:
+      - virtual_repository_config
+    description: |-
+      Configuration specific for a Remote Repository.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'description'
+        description: |-
+          The description of the remote source.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'dockerRepository'
+        exactly_one_of:
+          - docker_repository
+          - maven_repository
+          - npm_repository
+          - python_repository
+        description: |-
+          Specific settings for a Docker remote repository.
+        properties:
+          - !ruby/object:Api::Type::Enum
+            name: 'publicRepository'
+            exactly_one_of:
+              - public_repository
```

Comment:
exactly_on_of is not working here. Please see the generated downstream filed https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7333-old..auto-pr-7333#diff-0e32fc3bcc52373d438d360711325e4c95ae5f37ea79b5958908c2e8213771b3R154. 

ExactlyOneOf: []string{}, is an empty array.

Do you want to make the field `publicRepository` required?  You can use required: true to do that.

Reply:
The field is a oneof and we expect to add more options per publicRepository enum.